### ISSUE TITLE
False positive for `PD011` when `pd` does not exist in module

### ISSUE THREAD

**Issue by danparizher** [astral-sh/ruff#3807]
Code Snippet:
```py
async def select_callback(
    self,
    select: discord.ui.Select,
    interaction: discord.Interaction,
) -> None:  # the function called when the user is done selecting options
    value = select.values[0]
    if value == "current_value":
        # do nothing
        await interaction.response.defer()
        return
```
`PD011` comes up on the line `value = select.values[0]`, but pd doesn't even exist in this module.

**Comment by charliermarsh** [comment]
Unfortunately this is hard to prevent. We can't rely on the presence of a `pd` import, since you can use Pandas in a given module without importing it (e.g., imagine you call a function, defined in another file, that returns a DataFrame, and call `.values` on that DataFrame).

**Comment by MichaReiser** [comment]
One idea that we could support is to detect the frameworks in a project (or configure them manually for the time being), and then enable/disable the appropriate rules based on that (or use that knowledge for preciser inference)

**Comment by souliane** [comment]
I also get a false positive with this minimal example:
```python
class Dummy:
    values: str


x = Dummy().values
```
```shell
❯ ruff --version         
ruff 0.6.8

❯ ruff check test_ruff.py
test_ruff.py:5:5: PD011 Use `.to_numpy()` instead of `.values`
  |
5 | x = Dummy().values
  |     ^^^^^^^^^^^^^^ PD011
  |
```
Given the simplicity of this code, I would expect ruff to rule out the possibility of `Dummy` being a pandas dataframe, but I understand that whether the code is simple or not does not impact the result. Or is this a similar but actually a different issue?

**Comment by MichaReiser** [comment]
This is related to https://github.com/astral-sh/ruff/issues/6432. I'll merge the two issues
